src/main/java/com/MyApp/budgetControl/api/CategoryController.java

```diff
@@ -0,0 +1,36 @@
+package com.MyApp.budgetControl.api;
+
+import com.MyApp.budgetControl.domain.expense.category.CategoryDTO;
+import com.MyApp.budgetControl.domain.expense.category.CategoryEntity;
+import com.MyApp.budgetControl.domain.expense.category.CategoryService;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/categories")
+public class CategoryController {
+
+  private final CategoryService categoryService;
+
+  @PostMapping
+  @ResponseStatus(HttpStatus.CREATED)
+  public CategoryEntity addNewCategory(@Valid @RequestBody CategoryDTO newCategory) {
```

Comment:
Do not return entity through API, change it to return DTO, You may follow the same convention as in Expenses controller to have request/response DTOs

Reply:
I added separate DTOs for request and response